components/arm/client.go

```diff
@@ -108,6 +111,9 @@ func (c *client) JointPositions(ctx context.Context, extra map[string]interface{
 	if err != nil {
 		return nil, err
 	}
+	if resp.Positions == nil {
+		c.logger.Warn("the joint positions are nil")
```

Comment:
More verbose, something like
```suggestion
		c.logger.Warn("arm joint positions are nil, check that you are passing non-empty joint positins when writing your driver")
```

or similar.

Also same comment for the other two warning.